### PR TITLE
ci: migrate lint job from CircleCI to GHA

### PR DESCRIPTION
**Pull request by josephperrott**
Migrate lint to Github Actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - main
+      - '[0-9]+.[0-9]+.x'
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions: {}
```
Comment (devversion) [inline .github/workflows/ci.yml:15]: can we use read-all here instead?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+defaults:
+  run:
+    shell: bash
```
Comment (devversion) [inline .github/workflows/ci.yml:19]: is this needed? Isn't the default bash all the time?

**Review by devversion — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Initialize environment
+        uses: angular/dev-infra/github-actions/npm/checkout-and-setup-node@852c3d1898775ec9fbc87213e519f16e78805b57
+        with:
+          cache-node-modules: true
+      - name: Install node modules
+        run: yarn install --immutable
```
Comment (devversion) [inline .github/workflows/ci.yml:30]: isn't `--immutable` a Yarn 2.x flag? we use 1.x here

File: .github/workflows/ci.yml (hunk 1 above)
Comment (josephperrott) [reply .github/workflows/ci.yml:15]: `permissions: {}` is actually no permissions, not read-all.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (josephperrott) [reply .github/workflows/ci.yml:19]: Bash is the default shell in linux, but not in windows.  So while this is just the lint job, eventually when windows is in here too we will need to declare the default shell.

Comment (devversion) [reply .github/workflows/ci.yml:19]: fair enough. I don't expect anything to run with Windows in the components repo though

**Review by devversion — APPROVED** [review]

**Pull request merged**